lib/config/dynamic/merger_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dynamic
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/vic/pkg/registry"
+)
+
+func TestWhitelistMerger(t *testing.T) {
+	var tests = []struct {
+		orig, other registry.Entry
+		res         registry.Entry
+		err         error
+	}{
+		{
+			orig:  registry.ParseEntry("10.10.10.10"),
+			other: registry.ParseEntry("10.10.10.10"),
+			res:   registry.ParseEntry("10.10.10.10"),
+		},
+		{
+			orig:  registry.ParseEntry("10.10.10.10"),
+			other: registry.ParseEntry("10.10.10.20"),
+		},
```

Comment:
Why would that be a weakened restriction? This would be just a union, which would expand the whitelist, but that does not necessarily mean its weakened.